exercises/allergies/example.cpp

```diff
@@ -0,0 +1,46 @@
+#include "allergies.h"
+
+#include <string>
+#include <map>
+#include <vector>
+#include <algorithm>
+
+namespace allergies
+{
+
+allergy_test::allergy_test(const int test_result)
+    : result(test_result)
+    , allergens({
+    	{"eggs", 1},
+	{"peanuts", 2},
+	{"shellfish", 4},
+	{"strawberries", 8},
+	{"tomatoes", 16},
+	{"chocolate", 32},
+    	{"pollen", 64},
+	{"cats", 128}
```

Comment:
This is a little weird. For every one of these objects you create an identical copy of this map object inside of it.

What you _could_ do instead is create a `const` global object in an anonymous namespace in this file. Then you'd only ever have one copy, and the objects would be small.

(Also, the formatting here is a bit messed up)